src/server/engine_shard_set.cc

```diff
@@ -862,10 +863,13 @@ void EngineShardSet::TEST_EnableCacheMode() {
 
 ShardId Shard(string_view v, ShardId shard_num) {
   bool has_hashtags = false;
-  if (ClusterConfig::IsEnabledOrEmulated()) {
+
+  static const auto round_robin_prefix = absl::GetFlag(FLAGS_shard_round_robin_prefix);
```

Comment:
That's not a great idea.
Function-static variables have a built-in atomic into them, and so by "caching" the flag value in such a variable you don't save anything (flags are also protected by an atomic var). And this function is super hot path, so we don't want to use atomics here.
Fortunately we have `RoundRobinSharder::IsEnabled()` that caches this value in a thread-local and efficient manner :)